File: internal/config/path.go
package config

import (
	"fmt"
	"os"
	"path/filepath"
)

const (
	appConfigDirName  = "codestrike"
	appConfigFileName = "default.yaml"
)

// ResolvePath returns the config file path to use: flagValue if non-empty,
// otherwise the OS default user config location (codestrike/default.yaml
// under os.UserConfigDir()).
func ResolvePath(flagValue string) (string, error) {
	if flagValue != "" {
		return flagValue, nil
	}

	dir, err := os.UserConfigDir()
	if err != nil {
		return "", fmt.Errorf("determining user config directory: %w", err)
	}

	return filepath.Join(dir, appConfigDirName, appConfigFileName), nil
}

File: src/config/path.ts
import * as fs from 'fs';
import * as path from 'path';
import * as os from 'os';

const APP_CONFIG_DIR_NAME = 'codestrike';
const APP_CONFIG_FILE_NAME = 'default.yaml';

export function resolvePath(flagValue?: string): string {
  if (flagValue) return flagValue;
  return path.join(getUserConfigDir(), APP_CONFIG_DIR_NAME, APP_CONFIG_FILE_NAME);
}

function getUserConfigDir(): string {
  switch (process.platform) {
    case 'darwin':
      return path.join(os.homedir(), 'Library', 'Application Support');
    case 'win32':
      return process.env.APPDATA || path.join(os.homedir(), 'AppData', 'Roaming');
    default:
      return process.env.XDG_CONFIG_HOME || path.join(os.homedir(), '.config');
  }
}
